gschem/src/x_autonumber.c

```diff
@@ -342,7 +342,7 @@ gint autonumber_match(AUTONUMBER_TEXT *autotext, OBJECT *o_current, gint *number
   if (g_str_has_suffix (str,"?")) {
     isnumbered = 0;
     /* There must not be any character between the "?" and the searchtext */
-    if (strlen(str) != len+1)
+    if ((gint) strlen(str) != len+1)
```

Comment:
`len` comes from a call to `strlen()`, so it should be declared as `size_t len` instead of `int len`.  Then this cast would not be necessary, because the comparison would have `size_t` on both sides.